### PR TITLE
Test scan start/stop

### PR DESCRIPTION
**Pull request by paul-szczepanek-arm**
Test that scan and stop scan always succeeded and the resulting state is correct in the end.

### CODE REVIEW

File: ble-cliapp/docs/command-list.md [hunk 1]
```diff
@@ -568,6 +568,16 @@ connection. Will not drop a connection if it already has been established.
 * modeled after: `Gap::disconnect`
 
 
+### isRadioActive
+
+* invocation: `gap isRadioActive`
```
Comment (pan-) [inline ble-cliapp/docs/command-list.md:573]: Fun fact: I think at some point it was!

File: test_suite/gap/test_start_stop_scan.py [hunk 2]
```diff
@@ -103,3 +104,31 @@ def test_scan_should_not_report_advertisement_when_peer_inactive(scanner: BleDev
...
+    scanner.gap.stopScan()
+    # this is async so we wait
+    sleep(0.5)
+    assert scanner.gap.isRadioActive().result is False
+
+    for i in range(10):
+        scanner.gap.stopScan()
+        scanner.gap.startScan(0, 'DISABLE', 0)
+
+    assert scanner.gap.isRadioActive().result is True
```
Comment (pan-) [inline test_suite/gap/test_start_stop_scan.py:126]: Is there enough time to start the scan between this command and the last one ?

File: ble-cliapp/docs/command-list.md (hunk 1 above)
Comment (paul-szczepanek-arm) [reply ble-cliapp/docs/command-list.md:573]: I am indeed amused

File: test_suite/gap/test_start_stop_scan.py (hunk 2 above)
Comment (paul-szczepanek-arm) [reply test_suite/gap/test_start_stop_scan.py:126]: Yes, starting is immediate as it immediately goes to pending which is enough as scan is not idle anymore.

**Review by pan- — APPROVED** [review]

**Comment by pan-** [comment]
I'm closing this PR and I'm opening a new one to let the CI build ble-cliapp

**Pull request closed without merging**